Source/WebGPU/WebGPU/CommandEncoder.mm

```diff
@@ -141,12 +141,12 @@ static MTLStoreAction storeAction(WGPUStoreOp storeOp)
     // FIXME: Implement this according to
     // https://gpuweb.github.io/gpuweb/#dom-gpucommandencoder-begincomputepass.
 
-    if (descriptor.timestampWriteCount && !m_device->hasFeature(WGPUFeatureName_TimestampQuery))
+    if (descriptor.timestampWrites && !m_device->hasFeature(WGPUFeatureName_TimestampQuery))
         return false;
 
-    for (uint32_t i = 0; i < descriptor.timestampWriteCount; ++i) {
-        const auto& timestampWrite = descriptor.timestampWrites[i];
-        if (timestampWrite.queryIndex >= fromAPI(timestampWrite.querySet).count())
+    if (descriptor.timestampWrites) {
+        const auto& timestampWrite = *descriptor.timestampWrites;
+        if (timestampWrite.beginningOfPassWriteIndex >= fromAPI(timestampWrite.querySet).count() || timestampWrite.endOfPassWriteIndex >= fromAPI(timestampWrite.querySet).count())
```

Comment:
There's a lot going on here. Should the `fromAPI(...)` be split out into a local variable?

```
auto querySetCount = fromAPI(timestampWrite.querySet).count();
if (timestampWrite.beginningOfPassWriteIndex >= querySetCount || timestampWrite.endOfPassWroteIndex >= querySetCount)
```